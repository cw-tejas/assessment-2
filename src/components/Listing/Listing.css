.grid-container {
    display: flex;
    flex-direction: column;
    padding: 0;
}

.sort-tool {
    width: 100%;
}

#listing {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-gap: 20px;
    width: 100%;
    overflow: scroll;
}