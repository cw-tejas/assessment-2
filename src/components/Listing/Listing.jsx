import React, { useEffect } from 'react'
import './Listing.css'
import Card from '../Card/Card'


const Listing = ({filter, cars, handleSortChange}) => {    

  return (
    <div className='grid-container'>
        <div className='sort-tool'>
          <span>Sort By: </span>
          <select onChange={handleSortChange} value={filter.sort} name="cars" id="cars">
            <option value="best-match">Best Match</option>
            <option value="price-low-high">Price - Low to High</option>
            <option value="price-high-low">Price - High to Low</option>
          </select>
        </div>
        <div className="grid-item" id='listing'>
          {
            cars.map((car) => {
              return (
                <Card key={car.profileId} car={car} />
              )
            })
          }  
        </div>
    </div>
  )
}

export default Listing