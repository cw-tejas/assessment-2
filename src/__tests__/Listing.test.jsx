import { render, fireEvent, screen } from "@testing-library/react";
import Listing from "../components/Listing/Listing";

const carsMock = [
  { profileId: "2", carName: "Car B", km: "5000", fuel: "Diesel", cityName: "City B", price: "1500000", imageUrl: "car-b.jpg", priceNumeric: "15000000" },
  { profileId: "1", carName: "Car A", km: "2000", fuel: "Petrol", cityName: "City A", price: "1000000", imageUrl: "car-a.jpg", priceNumeric: "10000000" },
  { profileId: "3", carName: "Car C", km: "10000", fuel: "Electric", cityName: "City C", price: "2000000", imageUrl: "car-c.jpg", priceNumeric: "20000000" },
];

const carsMockAscending = [
  { profileId: "1", carName: "Car A", km: "2000", fuel: "Petrol", cityName: "City A", price: "1000000", imageUrl: "car-a.jpg", priceNumeric: "10000000" },
  { profileId: "2", carName: "Car B", km: "5000", fuel: "Diesel", cityName: "City B", price: "1500000", imageUrl: "car-b.jpg", priceNumeric: "15000000" },
  { profileId: "3", carName: "Car C", km: "10000", fuel: "Electric", cityName: "City C", price: "2000000", imageUrl: "car-c.jpg", priceNumeric: "20000000" },
];

describe("Listing Component", () => {
  const handleSortChange = jest.fn();
  
  it("renders car listings", () => {
    render(<Listing cars={carsMock} filter={{}} handleSortChange={handleSortChange} />);
    expect(screen.getByText(/Car A/i)).toBeInTheDocument();
    expect(screen.getByText(/Car B/i)).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Listing cars={carsMock} filter={{}} handleSortChange={handleSortChange} />);
    expect(asFragment).toMatchSnapshot();
  });

  it("handles sort option change", () => {
    render(<Listing cars={carsMock} filter={{ sort: "best-match" }} handleSortChange={handleSortChange} />);
    const selectElement = screen.getByRole("combobox");

    fireEvent.change(selectElement, { target: { value: "price-low-high" } });
    console.log(selectElement.value);

    expect(handleSortChange).toHaveBeenCalledTimes(1);
  });
});
