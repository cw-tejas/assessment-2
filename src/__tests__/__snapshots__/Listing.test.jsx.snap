// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing Component matches snapshot 1`] = `[Function]`;
